components/dash-core-components/src/fragments/DatePickerSingle.react.js

```diff
@@ -10,23 +10,60 @@ import convertToMoment from '../utils/convertToMoment';
 export default class DatePickerSingle extends Component {
     constructor() {
         super();
+        this.propsToState = this.propsToState.bind(this);
         this.isOutsideRange = this.isOutsideRange.bind(this);
         this.onDateChange = this.onDateChange.bind(this);
         this.state = {focused: false};
     }
 
-    isOutsideRange(date) {
-        const {max_date_allowed, min_date_allowed, disabled_days} =
-            convertToMoment(this.props, [
+    propsToState(newProps, force = false) {
+        const state = {};
+
+        if (
+            force ||
+            newProps.max_date_allowed !== this.props.max_date_allowed
+        ) {
+            state.max_date_allowed = convertToMoment(newProps, [
                 'max_date_allowed',
+            ]).max_date_allowed;
+        }
+
+        if (
+            force ||
+            newProps.min_date_allowed !== this.props.min_date_allowed
+        ) {
+            state.min_date_allowed = convertToMoment(newProps, [
                 'min_date_allowed',
+            ]).min_date_allowed;
+        }
+
+        if (force || newProps.disabled_days !== this.props.disabled_days) {
+            state.disabled_days = convertToMoment(newProps, [
                 'disabled_days',
-            ]);
+            ]).disabled_days;
+        }
+
+        if (Object.keys(state).length) {
+            this.setState(state);
+        }
+    }
 
+    UNSAFE_componentWillReceiveProps(newProps) {
+        this.propsToState(newProps);
+    }
+
+    UNSAFE_componentWillMount() {
```

Comment:
Don't have to do this now as it's bigger than just this bug, but instead of using the `UNSAFE` methods it could be worthwhile to refactor the component to functional and `useMemo` for the values that needs mapping.

Reply:
So I need to rewrite the component into a functional one? Or leave it as is for now?

Reply:
You can leave it as is.